docs/DeveloperGuide.md

```diff
@@ -154,6 +154,47 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Sort Persons
+
+The sort persons feature allows the user to sort persons by their name, date of birth or gender. Sorting is performed as part of `listPersonsCommand#execute()`.
+
+The sorting operation is exposed in the `Model` interface as `Model#sortPersons()` which calls `AddressBook#sortPersons()` which in turn calls `UniquePersonList#sort()` to sort the underlying `ObservableList<Person>`.
+
+The following sequence diagram illustrates the relevant sorting method calls for the command `listPersons s/n` with a `SortField` object `s`:
+
+![SortPersonsSequenceDiagram](images/SortPersonsSequenceDiagram.png)
+
+
+The field to sort persons by is encapsulated within the `SortField` class. `SortField` stores the sort field enumeration (`SortFieldType`) as well as the `Comparator` object that defines the sorting logic.
+
+The following class diagram shows how the `SortField` class integrates with the other components:
+
+![SortFieldClassDiagram](images/SortFieldClassDiagram.png)
```

Comment:
It may be better to present the SortField class and its class diagram before the sequence diagram, as I was initially confused as to where the SortField class came from.